### ISSUE TITLE
Decrease time-to-first-result for checks and transforms

### ISSUE THREAD

**Issue by robredpath** [open-contracting/kingfisher#297]
- 1.1
- Validation

**Comment by robredpath** [comment]
We've talked about using Redis for this

**Comment by odscjames** [comment]
Update on this ticket:

We have done well at this looking over the total months long back to back sprints; we have moved from a situation where you couldn't run checks until ALL fetch's have finished, to a situation where checks can start as soon as data arrives in the database. 

We have done ok at this over just the last sprint; we have added a Redis queue. Messages are sent thru the queue on any new data, and workers process checks:

Concerns and future work:

General stability of the Redis queue. We are starting to add both Sentry and normal Python Logs to Kingfisher, so we can monitor this.

We have both the Redis workers and the Check cron job running. The purpose of this is that if the Redis workers die for any reason, the Cron job will still pick up and fix problems later. But they may clash, or duplicate (thus wasted) work and it's not ideal.

The Redis workers don't currently process any transforms. There is no particular reason why they shouldn't, but note concern above.

The messages sent thru the queue currently contain just the collection ID, so when the checks process starts it checks all data in that collection. It might be better if the queue contained more ID's of the specific data added (file ID's and or file item IDs) so the checks and transforms code can be more targeted. This allows more parallelism with less crash worries - if 2 files are uploaded and there are 2 workers, currently they may clash and the work of one will be duplicating the work of others, and thus wasted. If we fixed this we could do parallelism. 

(But, general note, fixing https://opendataservices.plan.io/issues/19874 might be the biggest gain in checks speed we can do right now. Certainly last weekend we had a situation where there was lots of checks work to be done (about 10 days, by estimate), and so sorting out ways of triggering it faster for time-to-first-result was NOT the bottle neck and concern.)
